webpack/JobInvocationDetail/CheckboxesActions.js

```diff
@@ -0,0 +1,195 @@
+import {
+  Button,
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  MenuToggle,
+  Tooltip,
+} from '@patternfly/react-core';
+import { OutlinedWindowRestoreIcon } from '@patternfly/react-icons';
+import EllipsisVIcon from '@patternfly/react-icons/dist/js/icons/ellipsis-v-icon';
+import axios from 'axios';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { foremanUrl } from 'foremanReact/common/helpers';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  getIdsArray,
+  templateInvocationPageUrl,
+  MAX_HOSTS_API_SIZE,
+} from './JobInvocationConstants';
+import { selectTemplateInvocation } from './JobInvocationSelectors';
+import OpenAllInvocationsModal, { PopupAlert } from './OpenAllInvocationsModal';
+
+export const CheckboxesActions = ({
+  allHostsIds,
+  areAllRowsSelected,
+  bulkParams,
+  failedCount,
+  isAllSelected,
+  jobID,
+}) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+  const [isOpenFailed, setIsOpenFailed] = useState(false);
+  const idsArray = getIdsArray(bulkParams, allHostsIds, isAllSelected);
+  const permissions = useSelector(
+    state => selectTemplateInvocation(state)?.permissions
+  );
+
+  const openLink = url => {
+    const newWin = window.open(url);
+
+    if (!newWin || newWin.closed || typeof newWin.closed === 'undefined') {
+      setShowAlert(true);
+    }
+  };
+
+  const fetchFailedHosts = async () => {
+    const { data } = await axios.get(
+      foremanUrl(`/api/job_invocations/${jobID}/hosts`),
+      {
+        params: {
+          per_page: MAX_HOSTS_API_SIZE,
+          search: `job_invocation.result = failed`,
+        },
+      }
+    );
+    return data?.results || [];
+  };
+
+  const handleOpenHosts = async (type = 'all') => {
+    if (type === 'failed') {
+      if (failedCount <= 3) {
```

Comment:
Why `3` here?

Reply:
My programming professor in university always insistent on constants with a descriptive name. Could we do that?